### ISSUE TITLE
Recurring event instances being ignored

### ISSUE THREAD

**Issue by djmtype** [jammeryhq/gridsome-source-google-calendar#2]
@noxify Recurring instances of an event are being ignored. 
By default, a recurring event will only be requested once. 
For example, if an event happens daily from Jan 1 - 15, it will only request the data for Jan 1, ignoring the 14 other days.

The Google Calendar API offers a `singleEvents` boolean option which is set to `false` by default. Should this be true in order to request its remaining instances?
https://developers.google.com/calendar/v3/reference/events/list

BTW,  when enabling `_raw`, I didn't see a `singleEvents` property.

**Comment by noxify** [comment]
Hey @djmtype 

thanks for reporting.

It seems `singleEvents` is just a property for the request.  Not for the response.

In the response you have a `recurrence` property, if the current event is "recurring".

Example:

```
"recurrence": [
    "RRULE:FREQ=WEEKLY;BYDAY=TH"
],
```

You can the details here: https://developers.google.com/calendar/v3/reference/events#resource

The plugin isn't limited to single events.

If you need an indicator or something else to know if an event is recurring, feel free to create an PR with the changes.

I will close this issue for now. Feel free to reopen it, if there is something :) 

Regards,
Marcus

**Comment by djmtype** [comment]
@noxify I guess I'm still a little unclear. How can each instance of these recurring events be returned as individual events when they aren't being requested that way? 

Here is a sample of a request for a recurring event: 
```
"node": {
            "summary": "Art of Rosie S",
            "start": {
              "date": "2021-01-01T10:00:00-05:00"
            },
            "_raw": {
              "recurrence": [
                "RRULE:FREQ=DAILY;UNTIL=20210215T045959Z;INTERVAL=1"
              ]
            }
          }
        }
```

This event happens every day until February 15, 2021. So, looking at the request using onegraph, this event (node) is listed ~45 times instead of just once. 

Side-by-side comparison between gridsome-source-google-calendar and onegraph: 
[https://cloudup.com/cXf65jZh1s0](https://cloudup.com/cXf65jZh1s0)

**Comment by noxify** [comment]
Maybe we have to change the logic how we fetch the data.

Maybe using `instances()` instead of `list()` could help.

https://developers.google.com/calendar/v3/reference/events/instances

If you want, give it a try - not sure if I have the time until 2021 to check this.

**Comment by djmtype** [comment]
Sorry Marcus, I didn't expect my one question to turn into such an adventure and then branch out into different paths. Thanks for everything. 🙏🙏🙏 I'm not gonna bug you until at least 2022!

**Comment by noxify** [comment]
No problem :) 

I will take a look if I have some time.

I will reopen the issue to make sure I will do it 🙈

**Comment by noxify** [comment]
Hey @djmtype 

i have added a new option `includeRecurringEvents` which allows you to get also the event records for recurring events.

**Comment by djmtype** [comment]
@noxify I should've posted my findings earlier. 

After my last comment, I decided to look at the request instead of the response. I then discovered your plugin was returning some instances of recurring events instead of just the initial instance. 

If the response happened to include a recurring event set to an earlier date prior to Now, it would either display one instance or none at all. 

For example, take this daily recurring event called "Winter Break" starting December 20 through January 20. Because the filter is set to show all upcoming events from Now, and today is January 5, it wouldn't show any recurring instances of "Winter Break."

On the other hand, if I didn't use the filter, it would show some recurring instances. However, the plugin still wasn't requesting all instances.

So that got me thinking, maybe Google's API has a default cap? Since my calendar has several hundred events, maybe the request had reached the end? (BTW, I don't know if this is true.)

So I went back, messing around with onegraph's calendar schema hoping something would spark. 
I saw Google's API or onegraph offered a `first` property value that took an integer, meaning "first x-amount of events." I specified 500 and saw it was delivering more events at the request time than before. Regardless, I had to set any `timeMin` value for this to work properly. Because Google's API uses `*.date` for the time of all-day events and `*.dateTime` for all others, that also needed to be reflected within the filter function's return. (I filtered out by the end time instead of start so all-day events would still be displayed on the page. I also used date-fns instead moment here.)

```js
filterEvents() {
  return this.$page.oneGraphCal.google.calendar.events.nodes.filter(
    (node) => {
       var eventDate = getTime(parseISO(node.end.date));
       var eventDateTime = getTime(parseISO(node.end.dateTime));
       var currentDate = Date.now();
       return (eventDate || eventDateTime) >= currentDate;
    }
);
```

I will pull your plugin for the update and see what happens. Thanks again!
